### PR TITLE
[CAS-957] Sync our ChatEvent structure with Backend Spec

### PR DESCRIPTION
**Pull request by bychkovdmitry**
https://stream-io.atlassian.net/browse/CAS-957

### 🎯 Goal

Sync our ChatEvent strcuture with Backend Spec

### 🛠 Implementation details

- Added `totalUnreadCount` and `unreadChannels` fields to the following events:
  - `notification.channel_truncated`
  - `notification.added_to_channel`
  - `notification.channel_deleted`

- Removed redundant events which can only be received by using webhooks:
  - `channel.created`
  - `channel.muted`
  - `channel.unmuted`
  - `channel.muted`
  - `channel.unmuted`

- Removed `watcherCount` field from the following events as they are not returned with the server response:
  - `message.deleted`
  - `message.read`
  - `message.updated`
  - `notification.mark_read`

 - Removed `user` field from the following events as they are not returned with the server response: 
  - `notification.channel_deleted`
  - `notification.channel_truncated`

- Made the `user` field in `channel.hidden` and `notification.invite_accepter` events non nullable.

- Added `NotificationInviteRejectedEvent`
- Added `member` field to the `NotificationRemovedFromChannel` event

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (CMS, cookbooks, tutorial)
- [x] Reviewers added

### 🎉 GIF

giphy (2)

### CODE REVIEW

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/events/ChatEvent.kt [hunk 1]
```diff
@@ -23,348 +22,446 @@ public interface UserEvent {
...
-    override val type: String,
-    @SerializedName("created_at") override val createdAt: Date,
-    override val cid: String,
-    @SerializedName("channel_type") val channelType: String,
-    @SerializedName("channel_id") val channelId: String,
-    override val user: User,
-    val message: Message?,
-    val channel: Channel,
-) : CidEvent(), UserEvent
+public interface HasChannel {
```
Comment (JcMinarro) [inline stream-chat-android-client/src/main/java/io/getstream/chat/android/client/events/ChatEvent.kt:25]: We had `CidEvent` interface for this kind of events

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/events/ChatEvent.kt [hunk 2]
```diff
@@ -23,348 +22,446 @@ public interface UserEvent {
...
+}
+
+public interface HasWatcherCount {
+    public val watcherCount: Int
+}
+
+public interface HasUnreadCounts {
+    public val totalUnreadCount: Int?
+    public val unreadChannels: Int?
+}
```
Comment (JcMinarro) [inline stream-chat-android-client/src/main/java/io/getstream/chat/android/client/events/ChatEvent.kt:54]: This interface looks weird. It said that has unreadCounts but the values are nullable

File: stream-chat-android-docs/src/main/java/io/getstream/chat/docs/java/Channels.java [hunk 3]
```diff
@@ -582,14 +581,6 @@ public void channelMute() {
                     List<ChannelMute> mutes = result.data().getUser().getChannelMutes();
                 }
             });
-
-            // Get updates about muted channels
-            client.subscribeFor(
-                    new Class[]{ChannelsMuteEvent.class},
-                    channelsMuteEvent -> {
```
Comment (JcMinarro) [inline stream-chat-android-docs/src/main/java/io/getstream/chat/docs/java/Channels.java:589]: Shouldn't we subscribe for `notification.channel_mutes_updated` and/or `notification.mutes_updated` instead? :thinking:

File: stream-chat-android-docs/src/main/kotlin/io/getstream/chat/docs/kotlin/Channels.kt [hunk 4]
```diff
@@ -559,11 +557,6 @@ class Channels(val client: ChatClient, val channelClient: ChannelClient) {
                         val mutes: List<ChannelMute> = user.channelMutes
                     }
                 }
-
-            // Get updates about muted channels
-            client.subscribeFor<ChannelsMuteEvent> { event ->
```
Comment (JcMinarro) [inline stream-chat-android-docs/src/main/kotlin/io/getstream/chat/docs/kotlin/Channels.kt:564]: Same here

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/events/ChatEvent.kt (hunk 2 above)
Comment (bychkovdmitry) [reply stream-chat-android-client/src/main/java/io/getstream/chat/android/client/events/ChatEvent.kt:54]: I'm checking it... According to the spec these fields are not present in "message.new" and "notification.message_new" events, but I see them coming from the server with these events.

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/events/ChatEvent.kt (hunk 1 above)
Comment (bychkovdmitry) [reply stream-chat-android-client/src/main/java/io/getstream/chat/android/client/events/ChatEvent.kt:27]: Moved `channelType` and `channelId` to `CidEvent`:
```
public sealed class CidEvent : ChatEvent() {
    public abstract val cid: String
    public abstract val channelType: String
    public abstract val channelId: String
}
``` 
`HasChannel` interface now contains only full `Channel` object

File: stream-chat-android-docs/src/main/kotlin/io/getstream/chat/docs/kotlin/Channels.kt (hunk 4 above)
Comment (bychkovdmitry) [reply stream-chat-android-docs/src/main/kotlin/io/getstream/chat/docs/kotlin/Channels.kt:564]: Updated to `NotificationChannelMutesUpdatedEvent` event.

File: stream-chat-android-docs/src/main/java/io/getstream/chat/docs/java/Channels.java (hunk 3 above)
Comment (bychkovdmitry) [reply stream-chat-android-docs/src/main/java/io/getstream/chat/docs/java/Channels.java:589]: Updated to `NotificationChannelMutesUpdatedEvent` event.

**Pull request merged**